src/Features/Core/Portable/InheritanceMargin/Finders/BaseTypeSymbolsFinder.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.FindSymbols;
+using Microsoft.CodeAnalysis.FindSymbols.FindReferences;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.InheritanceMargin.Finders
+{
+    internal class BaseTypeSymbolsFinder : InheritanceSymbolsFinder
+    {
+        public static readonly BaseTypeSymbolsFinder Instance = new();
+
+        protected override Task<ImmutableArray<ISymbol>> GetAssociatedSymbolsAsync(
+            ISymbol symbol,
+            Solution solution,
+            CancellationToken cancellationToken)
+        {
+            var namedTypeSymbol = (INamedTypeSymbol)symbol;
+            if (namedTypeSymbol.BaseType == null)
+            {
+                // AllInterfaces are topologically sorted by default.
+                return Task.FromResult(namedTypeSymbol.AllInterfaces.CastArray<ISymbol>());
+            }
+
+            // Calculate indegree for all the symbols
```

Comment:
'indegree' isn't a term i'm very familiar with.  consider expanding on what's going on here, and why.

Reply:
Probably it should be called 
'Symbols Before a symbol', I will update the comments